src/pages/NFT/NFT.tsx

```diff
@@ -0,0 +1,71 @@
+//- React Imports
+import { FC } from 'react';
+
+//- Components Imports
+import HistoryList from '../../features/view-nft/HistoryList/HistoryList';
+import TokenHashInfo from '../../features/view-nft/TokenHashInfo/TokenHashInfo';
+import NFTMetrics from '../../features/view-nft/NFTMetrics/NFTMetrics';
+import DomainPreview from '../../features/domain-preview/DomainPreview';
+import Actions from '../../features/view-nft/Actions/Actions';
+
+//- Hooks Imports
+import { useDomainEvents } from '../../lib/hooks/useDomainEvents';
+import { useBidData } from '../../lib/hooks/useBidData';
+
+//- Library Imports
+import { Domain, DomainMetrics, TokenPriceInfo } from '@zero-tech/zns-sdk';
+
+//- Types Imports
+import { Metadata } from '../../lib/types/metadata';
+
+type NFTContainerProps = {
+	domain: Domain;
+	metrics: DomainMetrics;
+	domainMetadata: Metadata;
+	paymentTokenInfo: TokenPriceInfo;
+};
+
+const NFTContainer: FC<NFTContainerProps> = ({
+	domain,
+	metrics,
+	domainMetadata,
+	paymentTokenInfo,
+}) => {
+	const { data: bids } = useBidData(domain?.id);
+	const { data: domainEvents, isLoading: isEventDataLoading } = useDomainEvents(
+		domain?.id,
+	);
```

Comment:
We can move these queries to the subcomponents. `react-query` will ensure that duplicate queries aren't executed simultaneously. Very handy!

Reply:
Awesome - done :)